File: ArrayExercise/FIndMinPlatform/MinPlatform.java
package ArrayExercise.FIndMinPlatform;

public class MinPlatform {

    public static void main(String[] args) {
        double A[] = {9.00, 9.40, 9.50, 11.00, 15.00, 18.00};
        double D[] = {9.10, 12.00, 11.20, 11.30, 19.00, 20.00};
        int platform = (A.length>0)? 1:0;
        //Not correct Implementation a it will fail for
        //platform which became empty which were previously occupied
        for(int i=0;i<A.length;i++){
            if(A[i+1]<D[i]){
                platform++;
            }
        }
        System.out.println("Min platform required= "+platform);
    }
}

File: ArrayExercise/FIndMinPlatform/MinPlatform.py
def minPlatform(arrival, dep):
    i= 0
    j=0
    max_platform=0
    platform=0
    while(i < len(arrival) and j < len(dep)):
        if(arrival[i]<dep[j]):
            platform+=1
            if(platform>max_platform):
                max_platform=platform
            i+=1
        else:
            platform-=1
            j+=1
    return max_platform

a = [9.00, 9.40, 9.50, 11.00, 15.00, 18.00]
D = [9.10, 12.00, 11.20, 11.30, 19.00, 20.00];
print(minPlatform(a,D))
